Homework30/ClientSide/src/main/java/org/example/ClientHW.java

```diff
@@ -0,0 +1,70 @@
+package org.example;
+
+import java.io.*;
+import java.net.Socket;
+import java.util.Scanner;
+import java.util.logging.Logger;
+
+public class ClientHW {
+
+    private static final String SERVER_IP = "localhost";
+    private static final int SERVER_PORT = 3345;
+    private static final Logger LOGGER = Logger.getLogger(ClientHW.class.getName());
+
+
+Socket socket;
+    BufferedReader reader;
+    PrintWriter writer;
+    Scanner scanner;
+
+
+    public static void main(String[] args) {
+        ClientHW client = new ClientHW();
+        client.start();
+
+    }
+
+    public ClientHW() {
+        scanner = new Scanner(System.in);
+    }
+
+    public void start() {
+        try {
+            socket = new Socket(SERVER_IP, SERVER_PORT);
+            LOGGER.info("Connected to server.");
+
+            reader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            writer = new PrintWriter(socket.getOutputStream(), true);
+
+            Thread messageThread = new Thread(this::handleServerMessages);

```

Comment:
Would be better if client extends Runnable/Callable and do not hardcode thread-related logic here inside the client.